userspace/libsinsp/filterchecks.cpp

```diff
@@ -2142,6 +2142,14 @@ uint8_t* sinsp_filter_check_thread::extract_thread_cpu(sinsp_evt *evt, OUT uint3
 	return NULL;
 }
 
+// Some syscall sources, such as the gVisor integration, cannot match events to host PIDs and TIDs.
+// The event will retain the PID field which is consistent with the rest of sinsp logic, but it won't represent
+// a real PID and so it should not be displayed to the user.
+inline bool should_extract_xid(int64_t xid)
+{
+	return xid >= -1 && xid <= UINT32_MAX;
```

Comment:
Why also - 1 here? Is this valid in our logic?

Reply:
There may be cases in which we do return -1 if sinsp cannot determine the correct thread ID but I'm not sure. In any case, I don't want to introduce breaking changes :)